controllers/edgedeployment_controller.go

```diff
@@ -285,3 +358,25 @@ func mergeErrorMessages(errs []error) string {
 	}
 	return message
 }
+
+func SplitEdgeDevices(s []managementv1alpha1.EdgeDevice, x int) [][]managementv1alpha1.EdgeDevice {
```

Comment:
There is no need for this to be public (here). Maybe moving it to utils or specialized device utils would be an option? 
Also, there is nothing in the function that actually needs `EdgeDevice` so maybe  using `interface{}` in parameter and return value would be possible?

Reply:
It should be **s**plitEdgeDevices. I did not find a way to make it work with interface{}. All examples and docs I saw do not offer a solution.
If you have one please suggest.